src/platforms/common/swdptap.c

```diff
@@ -108,8 +121,9 @@ static uint32_t swdptap_seq_in_no_delay(const size_t clock_cycles)
 	for (size_t cycle = 0; cycle < clock_cycles; ++cycle) {
 		gpio_clear(SWCLK_PORT, SWCLK_PIN);
 		value |= gpio_get(SWDIO_IN_PORT, SWDIO_IN_PIN) ? 1U << cycle : 0U;
+		__asm__("" ::: "memory");
```

Comment:
Please insert a note here about this being a reordering barrier - same as in other places the trick is used.

Reply:
Added a comment. Both `swdptap_seq_in_no_delay()` and `swdptap_seq_out_no_delay()` need careful consideration in code influencing bitbanged timings, including effect of `gpio_set()/clear()` or `gpio_set_val()` and `gpio_get()`.
I would like to see this entire routine fit into ART cache of STM32F4, or at least not jump around thanks to all the `gpio_*` bits inlined.